Headers/Public/VLCTranscoder.h

```diff
@@ -0,0 +1,65 @@
+/*****************************************************************************
+ * VLCTranscoder.h: VLCKit.framework VLCTranscoder implementation
+ *****************************************************************************
+ * Copyright (C) 2018 Carola Nitz
+ * Copyright (C) 2018 VLC authors and VideoLAN
+ * $Id$
+ *
+ * Authors:  Carola Nitz <caro # videolan.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ *****************************************************************************/
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+* Transcoder delegate allows to be notified about transcoding state
+*/
+@protocol VLCTranscoderDelegate <NSObject>
+
+@optional
+/**
+ * Sent by the default notification center whenever the transcoder state has changed.
+ * \param aNotification
+ */
+- (void)transcoderFinishedSucessfully:(BOOL)success;
+
+@end
+
+/**
+ * Provides an object to convert a subtitle file and moviefile into one.
+ */
+@interface VLCTranscoder: NSObject
```

Comment:
Is this really a transcoder or a transcode? It seems to make sense to have a Transcoder to do transcode jobs but then it seems weird to have a delegate for the transcoder state and not the individual transcode jobs, how would you tell apart changes when you have multiple transcodes?